### ISSUE TITLE
[Advanced Select] Dropdown not closing in Chrome

### ISSUE THREAD

**Issue by oliverhaas** [htmlstreamofficial/preline#414]
### Summary

The advanced select dropdown stays open on Chrome when trying to close it.

### Steps to Reproduce

1. Go to https://preline.co/docs/advanced-select.html with Chrome (version see below).
2. Open a advanced select dropdown.
3. Try to close it again with unfocus or button click.

I have two similar but independent Ubuntu 22.04 systems with Chrome 127.0.6533.72 and a Windows system with Chrome 127.0.6533.73, all exhibit this behavior. Firefox and Chromium (126.0.X) do not seem to have this issue. Chrome version 126.0.X does not have the issue either.

EDIT: Checked out 128.0.X (unstable) as well, and the issue persists. So I don't think the problem will disappear on Chrome's end.

### Demo Link

https://preline.co/docs/advanced-select.html

### Expected Behavior

Dropdown should disappear again on unfocus, button click, etc.

### Actual Behavior

Dropdown stays open and glitched out, slightly shifted (margin changed).
I can see that the hidden class is not set, but some other attributes are changed.

### Screenshots

Normal opening behavior:
![image](https://github.com/user-attachments/assets/0bc2fe6f-59ef-4303-9938-82d31eeafd96)

But stays open after trying to close it:
![image](https://github.com/user-attachments/assets/c4a3f114-7d59-4ae3-b03e-c09bfa786d6b)

**Comment by oliverhaas** [comment]
I've found the cause of the problem.

At some point the HSSelect uses afterTransition() inside the close() function, and the afterTransition() function makes a check whether there is a transition or not

```
const afterTransition = (el: HTMLElement, callback: Function) => {
	const handleEvent = () => {
		callback();

		el.removeEventListener('transitionend', handleEvent, true);
	};
	const hasTransition =
		window.getComputedStyle(el, null).getPropertyValue('transition') !==
		(navigator.userAgent.includes('Firefox') ? 'all' : 'all 0s ease 0s');

	if (hasTransition) el.addEventListener('transitionend', handleEvent, true);
	else callback();
};
```

The value of `window.getComputedStyle(el, null).getPropertyValue('transition')` apparently changed from `'all 0s ease 0s'` to just `'all'` in Chrome 127.0.X (so just like Firefox now).

I don't quite understand why the callback isn't still working, but that's where I'm at now.

**Comment by jahaganiev** [comment]
Hey @oliverhaas - could you make a video record of the issue? At the time being, our team isn't able to reproduce the issue. Thanks!

**Comment by oliverhaas** [comment]
[Screencast from 26.07.2024 06:20:15.webm](https://github.com/user-attachments/assets/e302a9b6-9710-4bc4-b891-fc0ee780a2dc)

You can't see in the video that I'm furiously clicking in multiple placed to close the select ;).


I'm quite surprised that you can't reproduce the issue.

Are you on the current Chrome version (127.0.X)?
Did you go to https://preline.co/docs/advanced-select.html and try to open and close a select?

That's how I could reproduce it on 7 machines up to now (Windows, Mac, Ubuntu), and by three people independently. You can see your Chrome version in the menu (Help -> About Google Chrome). Don't forget to restart Chrome. The update just shipped 3 days ago and many people will not have restart their browser yet. If you want to make a quick call, I'm open to that as well (I'm in GMT+2).

**Comment by oliverhaas** [comment]
@jahaganiev To me it looks like that this will affect all Chromium based browser, and all websites using Preline HSSelect, and basically out of nowhere, so this is fairly serious. I'm not aware of anything I could be still missing, and would be happy to be wrong.

**Comment by f-tepel** [comment]
Hi @jahaganiev ,
same issue here. Update to latest Chrome version destroyed all our selects.

**Comment by olegpix** [comment]
@f-tepel @oliverhaas Hi,
We have released a hot-fix due to the Chrome update. Please update to Preline 2.4.1

**Comment by jahaganiev** [comment]
Hey @oliverhaas - once again, thanks for pointing out and reporting the issue with details. It helped our team quickly fix the issue. Appreciate it much!

**Comment by Mahdin-22** [comment]
Hi @jahaganiev
A few days ago I updated Chrome browser and today as I started working again, I realized that Advanced Selects are not closing anymore. I googled about it and I found myself here in a very new issue. I was so happy to see that this problem is solved so quickly and then as @olegpix suggested, I updated Preline but I had no change in select behavior. I updated all dependencies and still no progress. Is it so wired that I have the same issue in Firefox and MS Edge? Yes, that's true! And of course I had no chance to solve this by deleting cached files or using Incognito.
I really appreciate if you help me to solve this problem.
Thanks in advance!
